packages/trader/src/AppV2/Containers/Positions/positions.tsx

```diff
@@ -36,7 +38,11 @@ const Positions = observer(() => {
 
     React.useEffect(() => {
         setPositionURLParams(tabs[activeTab].id);
-        return () => onUnmount();
+
+        return () => {
+            const is_contract_details = history.location.pathname.startsWith('/contract/');
+            if (!is_contract_details) onUnmount();
```

Comment:
If it's NOT Contract Details page, then we reset filters by calling onUnmount. But if it's Contract Details then we want to preserve filters

Reply:
I have one query, when we click on markets or trade and then come back to positions is it expected to preserve the date

Reply:
I'm not sure if I understand you right, but if user is on markets or trade we should clean all filters (aka positions-store data)

Reply:
it looks fixed, please ignore my comment